onnx/defs/schema.cc

```diff
@@ -575,11 +591,32 @@ std::ostream& operator<<(std::ostream& out, const OpSchema& schema) {
   return out;
 }
 
-OpName_Domain_Version_Schema_Map& OpSchemaRegistry::map() {
+// Private method used by OpSchemaRegisterOnce and OpSchemaRegistry::map()
+OpName_Domain_Version_Schema_Map& OpSchemaRegistry::GetMapWithoutEnsuringRegistration() {
   static OpName_Domain_Version_Schema_Map map;
   return map;
 }
 
+OpName_Domain_Version_Schema_Map& OpSchemaRegistry::map() {  
+  auto& map = GetMapWithoutEnsuringRegistration();
+
+  class SchemasRegisterer {
+    public:
+      SchemasRegisterer() {
+        RegisterOnnxOperatorSetSchema();
+#ifdef ONNX_ML
+        RegisterOnnxMLOperatorSetSchema();
+#endif    
+      }
+  };
+
+  #ifndef ONNX_DISABLE_STATIC_REGISTRATION
```

Comment:
Can we remove this variant? It's not happening at initialization time already, what's the concern?

Reply:
This was intended to allow the static registry to be used by a caller which wants to constrain it to specific operator sets.  For example, to deprecate an older operator set or withhold others.

Reply:
This applies to your question below about custom schema. They may also implement ISchemaRegistry.

Reply:
Could you elaborate what's the purpose of constrain the registration to a specific operator sets? If it's for overriding an operator definition in the onnx core I'm pretty against it.

Reply:
No problem.  It's not actually intended for overriding ones that already exist.  One goal is for a caller to have more control over the contract it exposes to its own customers.  This naturally includes the operators that it supports. 

Another goal is to enable callers to handle certain operator sets outside the core library, based in part on its strategy for deploying updates.